### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Btslade**
Created documentation that can later be used to generate PDFS

### CODE REVIEW

File: docs/user_guides/Gufi for Administrators/sample.bib [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@article{greenwade93,
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Administrators/sample.bib:1]: Is this file needed?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 2]
```diff
@@ -0,0 +1,226 @@
+\documentclass{article}
+
+% Language setting
+% Replace `english' with e.g. `spanish' to change the document language
+\usepackage[english]{babel}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:5]: Is this needed?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+\usepackage{graphicx}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Coder's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:22]: Can you move the sections into different files so that the docs aren't giant main.tex files?

File: docs/user_guides/Gufi for Non Administrative users/main.tex [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+% Set page size and margins
+% Replace `letterpaper' with`a4paper' for UK/EU standard size
+\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
+
+% Useful packages
+\usepackage{amsmath}
+\usepackage{graphicx}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Common User's Guide}
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Non Administrative users/main.tex:16]: No need for `Common`

Comment (calccrypto) [reply docs/user_guides/Gufi for Non Administrative users/main.tex:16]: Same with the directory name - no need for `Non Administrative`

File: docs/user_guides/Gufi for Non Administrative users/main.tex [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+-t & sort by modification time, newest first\\
+-U & do not sort; list entries in directory order\\
+--delim \textless c\textgreater & delimeter separating output columns\\ 
+--in-memory-name \textless name\textgreater & Name of in-memory database when -R is used\\
+--nlink-width \textless chars\textgreater & Width of nlink column\\
+--size-width \textless chars\textgreater & Width of size column\\
+--user-width \textless chars\textgreater & Width of user column\\
+--group-width \textless chars\textgreater & width of group column
+\end{tabular}
+\caption{\label{tab:widgets}Flags and Functionality}
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Non Administrative users/main.tex:118]: `tab:widgets`?

File: docs/user_guides/Gufi for Non Administrative users/main.tex [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish. 
+
+\subsection{Background}
+This is accomplished by recreating the tree structure via indexing. Each directory contains an SQL database file that stores the metadata of the files as well as summary information for that directory and optionally summary information for the entire tree below that directory. 
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=0.5\textwidth]{gufi_structure.png}
+\caption{\label{fig:gufi\_structure}Layout and user interaction with GUFI}
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Non Administrative users/main.tex:31]: don't need `\`

File: docs/user_guides/Gufi For Coders/main.tex [hunk 7]
```diff
@@ -0,0 +1,226 @@
+\documentclass{article}
+
+% Language setting
+% Replace `english' with e.g. `spanish' to change the document language
+\usepackage[english]{babel}
+
+% Set page size and margins
+% Replace `letterpaper' with`a4paper' for UK/EU standard size
+\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:9]: Is this needed?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+
+% Useful packages
+\usepackage{amsmath}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:12]: Is this needed?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+\usepackage{graphicx}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Coder's Guide}
+\author{Braeden Slade}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:17]: Change to `GUFI Dev Team`

File: docs/user_guides/Gufi For Coders/main.tex [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+% Set page size and margins
+% Replace `letterpaper' with`a4paper' for UK/EU standard size
+\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
+
+% Useful packages
+\usepackage{amsmath}
+\usepackage{graphicx}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Coder's Guide}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:16]: `Developer's Guide`

**Comment by calccrypto** [comment]
Bring in the CMake stuff from 5772fce8eb6939d2a8379eb20cd2ecd7df3b325f and modify it to build the docs when building the code.

**Comment by calccrypto** [comment]
Don't use duplicate images. Have the LaTeX files reuse them

File: docs/user_guides/Gufi For Coders/main.tex [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish. 
+
+\subsection{Background}
+This is accomplished by recreating the tree structure via indexing. Each directory contains an SQL database file that stores the metadata of the files as well as summary information for that directory and optionally summary information for the entire tree below that directory.
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:26]: Ambiguous `This`.

File: docs/user_guides/Gufi for Administrators/main.tex [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: An Administrator's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish.
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Administrators/main.tex:23]: same intro as `Gufi for Coders`

File: docs/user_guides/Gufi for Administrators/main.tex [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+\title{GUFI: An Administrator's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish.  
+
+\subsection{Background}
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Administrators/main.tex:25]: same background as Gufi for Coders

File: docs/user_guides/Gufi for Non Administrative users/main.tex [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Common User's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish.
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Non Administrative users/main.tex:23]: same intro as Gufi for Coders

File: docs/user_guides/Gufi for Non Administrative users/main.tex [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish. 
+
+\subsection{Background}
+This is accomplished by recreating the tree structure via indexing. Each directory contains an SQL database file that stores the metadata of the files as well as summary information for that directory and optionally summary information for the entire tree below that directory.
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Non Administrative users/main.tex:26]: same background as Gufi for Coders

File: docs/user_guides/Gufi For Coders/main.tex [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+-w & open the database files in read-write mode instead of read only mode
+\end{tabular}
+\caption{\label{tab:widgets}Flags and Arguments}
+\end{table}
+
+\clearpage
+
+\subsection{Example Calls}
+
+gufi\_query -S "SELECT * FROM summary" ~/directory\_of\_root\_index
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:118]: Use [code blocks](https://www.overleaf.com/learn/latex/Code_listing) and/or `\texttt` for code and commandline examples.

**Review by calccrypto — CHANGES_REQUESTED** [review]

File: docs/user_guides/Gufi For Coders/main.tex [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+\title{GUFI: A Coder's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish. 
+
+\subsection{Background}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:25]: Why is this a subsection?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+
+\begin{table} [h]
+\centering
+\begin{tabular}{l|p{6cm}}
+Flag & Functionality \\\hline
+-h & help\\
+\hline
+-H & show assigned input values (debugging)\\
+\hline
+-E \textless SQL ent\textgreater & SQL for entries table \\
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:71]: https://tex.stackexchange.com/questions/2369/why-do-the-less-than-symbol-and-the-greater-than-symbol-appear-wrong-as

**Review by calccrypto — CHANGES_REQUESTED** [review]

File: docs/user_guides/Gufi For Coders/main.tex [hunk 19]
```diff
@@ -0,0 +1,226 @@
...
+\section{gufi\_query}
+
+\subsection{Outline}
+
+gufi\_query is used in order to query information from the generated databases that include all of the information about the indexed root directory and its sub-directories while also taking into account permissions 
+\subsection{Flags}
+
+\begin{table} [h]
+\centering
+\begin{tabular}{l|p{6cm}}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:65]: 6cm?

**Review by calccrypto — CHANGES_REQUESTED** [review]

File: docs/user_guides/Gufi For Coders/main.tex (hunk 19 above)
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:65]: 6cm?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 20]
```diff
@@ -0,0 +1,226 @@
...
+\subsection{Visualizing the Workflow}
+
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{gufi_query.png}
+\caption{\label{fig:gufi_query}Workflow of gufi\_query}
+\end{figure}
+
+\subsection{Processdir}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:132]: `\texttt{processdir}`

File: docs/user_guides/Gufi For Coders/main.tex [hunk 21]
```diff
@@ -0,0 +1,226 @@
...
+\includegraphics[width=1.0\textwidth]{gufi_dir2index.png}
+\caption{\label{fig:gufi_dir2index}gufi\_dir2index workflow}
+\end{figure}
+
+\subsection{Usage}
+gufi\_dir2index [flags] [directory to index] [location to place indexed directory]
+
+
+\section{Understanding Rollup}
+One of the predominant time savers in GUFI is the rollup feature which functions like so: \\
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:181]: `predominant time savers` -> `optimization`

File: docs/user_guides/Gufi For Coders/main.tex [hunk 22]
```diff
@@ -0,0 +1,226 @@
...
+
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{gufi_query.png}
+\caption{\label{fig:gufi_query}Workflow of gufi\_query}
+\end{figure}
+
+\subsection{Processdir}
+What you will quickly notice when tracing through the above diagram (Figure 3) the major functionality from this code comes from the processdir function. This function is where the -E -S and -T flags are processed. Therefore, this is where all of the SQL database accesses occur. This process is parallelized via a thread pool and is further optimized by a calculated rollupscore to ensure that we only descend into necessary sub-directories.
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:133]: > Therefore, this is where all of the SQL database accesses occur.

These queries access the index, but there are other queries that occur when aggregating.

File: docs/user_guides/Gufi For Coders/main.tex [hunk 23]
```diff
@@ -0,0 +1,226 @@
...
+\begin{figure} [h]
+\centering
+\includegraphics[width=0.5\textwidth]{gufi_structure.png}
+\caption{\label{fig:gufi\_structure}Layout and user interaction with GUFI}
+\end{figure}
+
+\clearpage
+
+\section{Understanding the data and how its stored}
+The data stored in these databases include filenames, access and creation dates, file attributes, and extended attributes. Depending on which table is viewed and what view is selected, there can be up to 59 attributes(columns) per entry(row)\\
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:37]: There **are** 59 attributes, some of which might be blank. More might be added in the future. Maybe change this to `there are currently 59 attributes`?

File: docs/user_guides/Gufi For Coders/main.tex [hunk 24]
```diff
@@ -0,0 +1,226 @@
...
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish. 
+
+\subsection{Background}
+This is accomplished by recreating the tree structure via indexing. Each directory contains an SQL database file that stores the metadata of the files as well as summary information for that directory and optionally summary information for the entire tree below that directory.
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=0.5\textwidth]{gufi_structure.png}
+\caption{\label{fig:gufi\_structure}Layout and user interaction with GUFI}
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:31]: `\label{fig:gufi\_structure}`

File: docs/user_guides/Gufi for Administrators/main.tex [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+-z \textless max\_level\textgreater & maximum level to go down to
+\end{tabular}
+\caption{\label{tab:widgets}Flags and Arguments}
+\end{table}
+\clearpage
+\subsection{Visualizing the Workflow}
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{gufi_dir2index.png}
+\caption{\label{fig:gufi dir2index} gufi\_dir2index workflow}
```
Comment (calccrypto) [inline docs/user_guides/Gufi for Administrators/main.tex:156]: \label{fig:gufi_dir2index}

**Comment by calccrypto** [comment]
Do you have files that future documentation editors can edit and use to generate updated images? Editing PNGs might result in weirdness due to antialiasing.

File: docs/user_guides/Gufi For Coders/main.tex [hunk 26]
```diff
@@ -0,0 +1,226 @@
...
+\end{figure}
+
+\subsection{Usage}
+gufi\_dir2index [flags] [directory to index] [location to place indexed directory]
+
+
+\section{Understanding Rollup}
+One of the predominant time savers in GUFI is the rollup feature which functions like so: \\
+\begin{itemize}
+  \item contents of the child's summary table are copied up with name column to include the parent directory
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:183]: Capitalization

File: docs/user_guides/Gufi For Coders/main.tex [hunk 27]
```diff
@@ -0,0 +1,226 @@
...
+  \item pentries view/table is dropped and a pentries table is created with the data inside the pentries view/table
+  \item copy child's pentries view into parent (regardless if view or table)
```
Comment (calccrypto) [inline docs/user_guides/Gufi For Coders/main.tex:185]: Capitalization

File: docs/latex/non_admin.tex [hunk 28]
```diff
@@ -0,0 +1,35 @@
...
+\usepackage{graphicx}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Common User's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\input{sections/introduction.tex}
```
Comment (calccrypto) [inline docs/latex/non_admin.tex:22]: switch to `\include` and you won't need `\clearpage`

File: docs/user_guides/Gufi For Coders/main.tex (hunk 2 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: It is not! I deleted it

File: docs/user_guides/Gufi For Coders/main.tex (hunk 3 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Modularized all of the documents

File: docs/user_guides/Gufi for Non Administrative users/main.tex (hunk 4 above)
Comment (Btslade) [reply docs/user_guides/Gufi for Non Administrative users/main.tex]: No more Non Administrative directory due to modularization.

File Names are now:
gufi_administrator.tex
gufi_developer.tex
gufi_user.tex

File: docs/user_guides/Gufi for Non Administrative users/main.tex (hunk 5 above)
Comment (Btslade) [reply docs/user_guides/Gufi for Non Administrative users/main.tex]: Removed! it was unnecessary and left over from an example document i was working off of

File: docs/user_guides/Gufi For Coders/main.tex (hunk 16 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Updated, I am using \texttt for my code examples

File: docs/user_guides/Gufi For Coders/main.tex (hunk 18 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Do you want me to implement that package in this stack exchange? The document currently compiles correctly anyway

File: docs/user_guides/Gufi For Coders/main.tex (hunk 19 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: This is to ensure that the formatting looks nice when the file is compiled into a pdf

File: docs/user_guides/Gufi For Coders/main.tex (hunk 20 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Done!

File: docs/user_guides/Gufi For Coders/main.tex (hunk 21 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Made the appropriate change

File: docs/user_guides/Gufi For Coders/main.tex (hunk 23 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Made the appropriate change

File: docs/user_guides/Gufi For Coders/main.tex (hunk 24 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Done!

**Comment by Btslade** [comment]
> Do you have files that future documentation editors can edit and use to generate updated images? Editing PNGs might result in weirdness due to antialiasing.

Yes, I have the draw.io files of all of these diagrams (except for the structure image I copied from the old documentation)

These files can be uploaded to the draw.io website and edited from there

Do you want me to upload these files into the repository? If so, where is the best place to do so?

File: docs/user_guides/Gufi For Coders/main.tex (hunk 27 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: Fixed

File: docs/latex/non_admin.tex (hunk 28 above)
Comment (Btslade) [reply docs/latex/non_admin.tex]: So for some of the formatting, i needed to use input, but I did go back and use include where appropriate

File: docs/user_guides/Gufi For Coders/main.tex (hunk 18 above)
Comment (calccrypto) [reply docs/user_guides/Gufi For Coders/main.tex]: I just thought there were too many `\textgreater`s in the text.

**Comment by calccrypto** [comment]
I guess put the draw.io files into the images directory as well

File: docs/user_guides/Gufi For Coders/main.tex (hunk 19 above)
Comment (calccrypto) [reply docs/user_guides/Gufi For Coders/main.tex]: Why 6cm and not 5 or 7? Is it even needed?

File: docs/latex/non_admin.tex (hunk 28 above)
Comment (calccrypto) [reply docs/latex/non_admin.tex]: What causes this? `\begin{tabular}{l|p{6cm}}`?

**Comment by calccrypto** [comment]
[rollup.png](https://github.com/mar-file-system/GUFI/blob/6b1ebead406db9a92560b827ab91c5b56b2bb411/docs/latex/images/rollup.png) is actually bottomup. diagram2 is missing.

**Comment by calccrypto** [comment]
Might want to rename querydb.png since there is also an executable called querydbs.

**Comment by Btslade** [comment]
> [rollup.png](https://github.com/mar-file-system/GUFI/blob/6b1ebead406db9a92560b827ab91c5b56b2bb411/docs/latex/images/rollup.png) is actually bottomup. diagram2 is missing.

1. The main line at the top is rollup, but rollup calls bottomup.c on the get record count step, and that's the majority of this diagram. I tried to highlight this by creating a green box with bottomup.c as a label to clarify this, maybe i should do something else to make it more clear?

2. I noticed in these diagrams, I am stating "check diagram2" Because in the draw.io interface, you can have multiple diagrams to one file. There is obviously no way someone reading the documentation would understand this, so I am going to go back, and clarify these by naming them differently. (in this case ill change it to "check ascend")

File: docs/user_guides/Gufi For Coders/main.tex (hunk 19 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: I will send pictures, If I don't use any formatting, the table  is messed up due to how much writing there is. Then I chose 6 as i felt it looked the best. I have images showing 5,6, and 7 for your reference.


without anything formatting

5cm

6cm

7cm

**Comment by calccrypto** [comment]
> The main line at the top is rollup,

The main line at the top is the main function of rollup, but really does not show any part of what it means to roll up. The core of rollup are the functions called during ascend.

File: docs/user_guides/Gufi For Coders/main.tex (hunk 19 above)
Comment (calccrypto) [reply docs/user_guides/Gufi For Coders/main.tex]: try replacing `p{6cm}` with `l`

**Comment by Btslade** [comment]
> The main line at the top is the main function of rollup, but does not show any part of the core rollup functionality. The core of rollup is the functions called during ascend.

Okay, I see what you are saying. I am going to have to figure out the best way to communicate what rollup.c is and where the actual functionality of rollup occurs in ascend

I will need to create an entirely seperate diagram that goes into detail about the check_children, can_rollup, dorollup and rollup functios.

File: docs/latex/gufi_administrator.tex [hunk 29]
```diff
@@ -0,0 +1,28 @@
+\documentclass{article}
+
+
+
```
Comment (calccrypto) [inline docs/latex/gufi_administrator.tex:4]: lots of unnecessary empty lines

Comment (calccrypto) [reply docs/latex/gufi_administrator.tex]: There's no need to remove every empty line. There's just no need for 2 and 3 empty lines everywhere.

**Comment by calccrypto** [comment]
[docs/latex/images/querydb_macro.png](https://github.com/mar-file-system/GUFI/blob/documentation/docs/latex/images/querydb_macro.png) Main does not call the macro.

**Comment by calccrypto** [comment]
Your .DS_Store files are getting pushed

**Comment by Btslade** [comment]
> [docs/latex/images/querydb_macro.png](https://github.com/mar-file-system/GUFI/blob/documentation/docs/latex/images/querydb_macro.png) Main does not call the macro.

I'm a little confused then. Because it looks like main calls processdir then processdir calls querydb

line 1023 in gufi_query.c (main function): ` QPTPool_enqueue(pool, i % in.maxthreads, **processdir**, mywork);`

line 582 in gufi_query.c (processdir function):  `querydb(dbname, db, "select name from sqlite_master where type=\'table\' and name='treesummary';", ta->print_callback_func, NULL, id, sqltsumcheck, recs);`

line 452 (what im assuming is the querydb macro being defined): `#define querydb(dbname, db, query, callback, obufs, id, ts_name, rc) `

Am i misunderstanding something?

**Comment by garygrider** [comment]
Processdir is sort of the dispatchable work unit or it use to be.  It’s a pretty important concept in the way gufi works.  Directories are the first class citizen in gufi indexing,  where files are just data 😊

From: Jason Lee ***@***.***>
Reply-To: mar-file-system/GUFI ***@***.***>
Date: Monday, July 11, 2022 at 9:27 AM
To: mar-file-system/GUFI ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [mar-file-system/GUFI] Documentation (PR #93)


You are correct with the call stack, but by that logic, main calls everything, which is too general. I would say that processdir calls querydb. However, in the picture I would say arguments and return to not specify who is calling querydb in the image since the gufi_query image already shows querydb being called.

File: docs/user_guides/Gufi For Coders/main.tex (hunk 19 above)
Comment (Btslade) [reply docs/user_guides/Gufi For Coders/main.tex]: When I do this, the lines are cut off
cuts off

**Review by calccrypto — APPROVED** [review]
